src/plugins/state/sagas.js

```diff
@@ -20,16 +25,35 @@ export function* setCanvas(action) {
 
   for (let windowId of windowIds) {
     let payload;
-    const canvas = yield select(getCanvases, { windowId });
-    const canvasId = canvas[0].id;
-    const maps = getMaps(canvas[0].iiifImageResources);
-    let images = getImages(canvas[0].iiifImageResources);
+    const canvas = yield select(getCurrentCanvas, { windowId });
+    const canvasId = canvas.id;
+    const maps = getMaps(canvas.iiifImageResources);
+    let images = getImages(canvas.iiifImageResources);
 
-    payload = reduceLayers(images, maps, excluded_maps);
-    yield put(updateLayers(windowId, canvasId, payload));
+    // length of one image implies no choices...
+    // only if the length is above one do we want to toggle layer visibility
+    if (images.length > 1) {
```

Comment:
You can have multiple images on a avanvas, see this recipe for example: https://iiif.io/api/cookbook/recipe/0036-composition-from-multiple-images/

in our own data we probably won't have a secanrio where the plugin is triggered and there's multiple images per canvas, but might be worth checking explicitly for choice

Reply:
OK